src/program/latex/mod.rs

```diff
@@ -0,0 +1,1600 @@
+    /// ```
+    pub fn get_command(command: Self) -> String {
+        match command {
+            Self::Lstick(wire) => format(format_args!(r#"\lstick{{\ket{{q_{{{wire}}}}}}}"#)),
+            Self::Rstick(wire) => format(format_args!(r#"\rstick{{\ket{{q_{{{wire}}}}}}}"#)),
+            Self::Gate(name) => format(format_args!(r#"\gate{{{name}}}"#)),
+            Self::Phase(symbol) => format(format_args!(r#"\phase{{{symbol}}}"#)),
+            Self::Super(script) => format(format_args!(r#"^{{\{script}}}"#)),
+            Self::Qw => r"\qw".to_string(),
+            Self::Nr => r"\\".to_string(),
+            Self::Meter(wire) => format(format_args!(r#"\meter{{{wire}}}"#)),
+            Self::Ctrl(wire) => format(format_args!(r#"\ctrl{{{wire}}}"#)),
+            Self::Targ => r"\targ{}".to_string(),
+            Self::Control => r"\control{}".to_string(),
+            Self::Swap(wire) => format(format_args!(r#"\swap{{{wire}}}"#)),
+            Self::TargX => r"\targX{}".to_string(),
+        }
+    }
+}
+
+/// Types of parameters passed to commands.
+#[derive(Debug)]
+pub enum Parameter {
+    /// Symbolic parameters
+    Symbol(Symbol),
+}
+
+impl Clone for Parameter {
+    fn clone(&self) -> Parameter {
+        match self {
+            Parameter::Symbol(symbol) => match symbol {
+                Symbol::Alpha => Parameter::Symbol(Symbol::Alpha),
+                Symbol::Beta => Parameter::Symbol(Symbol::Beta),
+                Symbol::Gamma => Parameter::Symbol(Symbol::Gamma),
+                Symbol::Phi => Parameter::Symbol(Symbol::Phi),
+                Symbol::Pi => Parameter::Symbol(Symbol::Pi),
+                Symbol::Text(text) => Parameter::Symbol(Symbol::Text(text.to_string())),
+            },
+        }
+    }
+}
+
+/// Supported Greek and alphanumeric symbols.
+#[derive(Debug)]
+pub enum Symbol {
+    Alpha,
+    Beta,
+    Gamma,
+    Phi,
+    Pi,
+    Text(String),
+}
+
+impl Symbol {
+    /// Returns the supported Symbol variant from text otherwise stores the
+    /// unsupported symbol as text in the Text variant.
+    ///
+    /// # Arguments
+    /// `text` - a String representing a greek or alaphanumeric symbol
+    pub fn match_symbol(text: String) -> Symbol {
+        if text == "alpha" {
+            Symbol::Alpha
+        } else if text == "beta" {
+            Symbol::Beta
+        } else if text == "gamma" {
+            Symbol::Gamma
+        } else if text == "phi" {
+            Symbol::Phi
+        } else if text == "pi" {
+            Symbol::Pi
+        } else {
+            Symbol::Text(text)
+        }
+    }
+    /// Returns the LaTeX String for a given Symbol variant.
+    ///
+    /// # Arguments
+    /// `symbol` - A Symbol variant.
+    ///
+    /// # Examples
+    /// ```
+    /// use quil_rs::program::latex::{Parameter, Symbol};
+    /// let alpha = Symbol::get_symbol(
+    ///     &Parameter::Symbol(Symbol::Alpha)
+    /// );
+    /// ```
+    pub fn get_symbol(symbol: &Parameter) -> String {
```

Comment:
This is confusing to me. There are three types involved: `&Parameter`, `String`, and `Symbol`. The function is named `get_symbol` but returns `String`.

I think this would be tidier as a `impl ToString` on _both_ `Parameter` and `Symbol`. (The `Parameter` impl would call the `Symbol` impl.)

Reply:
Great solution! That really resolved some of the confusion.

Reply:
@Shadow53 I gotta remember that tip. Ty